shared-module/src/components/TopicNavigation.tsx

```diff
@@ -118,7 +118,7 @@ const TopicNavigation: React.FC<TopicNavigationProps> = () => {
                     href={`#${id}`}
                     onClick={(e) => {
                       e.preventDefault()
-                      document.querySelector(`#${id}`)?.scrollIntoView({
```

Comment:
I don't think that works either:
![image](https://user-images.githubusercontent.com/1922896/162171047-a1232acf-2502-4959-a5e1-b3991258db7a.png)

Reply:
Works on my machine. :-)

Reply:
This should work:
```js
document.getElementById(id)?.scrollIntoView({
```

What you have now should never work because the querySelector api requires one to use CSS selectors which need the `#` prefix to tell that we're looking for an id.

The original solution actually worked as long as the first character of the UUID was not a number. (See https://stackoverflow.com/questions/20306204/using-queryselector-with-ids-that-are-numbers)

Reply:
Oh, that's confusing

Reply:
Even though this is a different feature, I think it's still wrong.

Reply:
Yea, I got it working by passing to `querySelector(`[id='${id}']`)`
Fixed aswell on this component, but now I need to figure out how to cut off the timeouts for other components once the scrolling succeeds, as this gets quite quickly annoying:
```ts
  useEffect(() => {
    if (typeof window != "undefined" && window.location.hash) {
      const selector = window.location.hash.substring(1)
      setTimeout(() => {
        tryToScrollToSelector(selector)
      }, 100)
      setTimeout(() => {
        tryToScrollToSelector(selector)
      }, 500)
      setTimeout(() => {
        tryToScrollToSelector(selector)
      }, 1000)
      setTimeout(() => {
        tryToScrollToSelector(selector)
      }, 2000)
    }
  }, [path])
```

Reply:
Got it working with returning boolean from tryToScrollToSelector.

Reply:
The multiple scrolls are to prevent layout shifts from breaking the scroll. We cannot get rid of layout shifts shortly after loading the page because we don't know the height of exercise iframes before loading them.

Maybe there should be comment there to explain this tradeoff?